File: zig/src/render/footnoteRefRenderer.zig
const std = @import("std");
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const RendererState = @import("../types/RendererState.zig").RendererState;
const Renderer = @import("../types/Renderer.zig").Renderer;

pub fn render(node: *const MarkdownNode, state: *RendererState, decode: ?bool) void {
    _ = decode;
    if (node.info) |info| {
        state.footnoteRefs.put(info, node) catch unreachable;
    }
}

pub const footnoteRefRenderer = Renderer{
    .name = "footnote_ref",
    .render = render,
};

File: web/src/render-console/footnoteRefRenderer.ts
import type MarkdownNode from "../types/MarkdownNode";
import type Renderer from "../types/Renderer";
import type RendererState from "../types/RendererState";

const renderer: Renderer = {
	name: "footnote_ref",
	render,
};
export default renderer;

function render(node: MarkdownNode, state: RendererState): void {
	if (node.info !== undefined) {
		state.footnoteRefs[node.info] = node;
	}
}
